test/interop/C/llvm/exportArray/arrayArg_queriedDomain.chpl

```diff
@@ -0,0 +1,4 @@
+export proc foo(x: [?D] int) {
```

Comment:
There's quite a lot of tests here. I'm hoping that in the future, we'll remove the `test/interop/C/llvm` directory and `test/interop/C` will cover the LLVM or C case depending on if the test run is already using `--llvm`. Otherwise it seems like we are making it harder for ourselves with all the duplication. 

Generally speaking I'd be happier with `test/interop/C` having skipifs for the tests that don't work yet with `--llvm` but the solution in this PR OK for the moment.

Reply:
This goes back to the issue I mentioned in our LLVM deep dive - having the library compilation in precomp files instead of .compopts files on the .chpl file itself means that passing `--llvm` to start_test doesn't impact how the library files are built.  And having the C compilation and execution be part of a prediff on the .chpl file instead seemed like it would tie us to testing one particular C compiler (instead of allowing us to check linking to the library using icc/cce/etc.).  I agree it's not ideal and I think there's some retooling we'll want to do on start_test but that seemed bigger than the scope of the PR

Reply:
I opened #12507